### DISCUSSION TITLE
Any plans to enable network client access based on the MySQL and/or Postgresql wire protocols?

### DISCUSSION THREAD

**Discussion by stephanbuys** [gluesql/gluesql#323] [Ideas]
This is a really cool project, recently I was speaking to colleagues about creating a SQL "proxy" for a project. The client software only supports Postgresql or MySQL and we were thinking of projects like Materialize (https://github.com/MaterializeInc/materialize) or Noria (https://github.com/mit-pdos/noria-mysql) for inspiration.

**Comment by panarch** [comment]
Thanks :)
Providing network client access is possible to implement, but I don't have a plan to work on it for now.

Main short-and-mid term goal of this project is for providing many SQL databases as possible.
There are lots of difference storage & environment which could be good to support SQL.
However, making SQL database is quite expensive task, so there exists only few SQL databases available to use in the market.
Here GlueSQL's main goal is reducing implementation cost of SQL database extremely, then many programmers including us can easily add SQL layer to their own storages.

Daemon process based DMBS require more effort to implement but I don't have enough resource for now.
I'm currently focusing on providing SQL databases which don't require standalone process.
Currently, the only reference storage `SledStorage` is embedded SQL database like SQLite3.
* In-memory storage
* IndexedDB storage
* CSV storage
* ... (Redis? .. I've not decided yet)
are also on the plan, soon you will be able to try those.

GlueSQL is kind of... storage "proxy", it provides almost same "SQL" to access different storages from in-memory, CSV, Redis, `sled`, .. to big data.

**Comment by KyGost** [comment]
I've been planning to solve this quite simply using ODBC; there's quite a nice ODBC crate already:
https://docs.rs/odbc/latest/odbc/

As such all that would be needed is a storage module.

Edit: ***Somehow*** I managed to misread MySQL as MSSQL...
